compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -67,25 +67,30 @@ static SmallVector<Value> getInputOperands(linalg::LinalgOp &linalgOp) {
   return operands;
 }
 
-/// Utility to fetch operands from the defining ops of LinalgOp's input
-/// operands. For example, we want to fetch the input operand of %pack0 and
-/// %pack1 as shown in below, and promote them to memory.
-/// %pack0 = tensor.pack % arg0
-/// %pack1 = tensor.pack % arg1
-/// %pack2 = tensor.pack % pack0
-/// %pack3 = tensor.pack % pack1
-/// %generic = linalg.generic ins(%pack2, %pack3)
-static FailureOr<SmallVector<Value>> getOperandsFromDefOp(
-    linalg::LinalgOp &linalgOp) {
+/// Utility to fetch pack operands at a specified depth from the LinalgOp's
+/// input operands.
+static FailureOr<SmallVector<Value>> getPackOperands(linalg::LinalgOp linalgOp,
+                                                     uint32_t depthLevel) {
   SmallVector<Value> operands;
-  for (Value input : linalgOp.getDpsInputs()) {
-    auto defOp = input.getDefiningOp<tensor::PackOp>();
+  for (auto input : llvm::enumerate(linalgOp.getDpsInputs())) {
+    uint32_t currentLevel{0};
+    Operation *currentOp = input.value().getDefiningOp();
+    while (currentLevel < depthLevel && currentOp != nullptr) {
+      if (dyn_cast<tensor::PackOp>(currentOp)) {
+        currentLevel++;
+        if (currentLevel == depthLevel) break;
+      }
+      currentOp = currentOp->getOperand(0).getDefiningOp();
+    }
     // The defining op has to be a pack op, fail otherwise.
-    if (!defOp) {
-      return failure();
+    if (!currentOp) {
+      return linalgOp.emitOpError()
+             << "operand #" << input.index() << " only has pack ops to depth "
+             << currentLevel << ", but request is for a depth " << depthLevel
+             << " pack op.";
```

Comment:
I don't know if it is easy to add a test for this.

Reply:
It's here: https://github.com/nod-ai/iree-amd-aie/pull/972/files#diff-08dc5392bd52ce08c6e138c89646fbccbec2f0c782d79323431ea1322919539dR102